core/app/models/spree/return_authorization.rb

```diff
@@ -61,6 +62,10 @@ def can_cancel_return_items?
 
     private
 
+      def authorize_return
+        order.authorize_return! unless order.nil?
```

Comment:
Yeah, there was this discussion already here https://github.com/spree/spree/issues/5636. Somewhere there is my comment as well, which overlaps what you said about the industry moving into domain objects.